optima/programs.py

```diff
@@ -16,6 +16,18 @@
 coveragepars=['numtx','numpmtct','numost','numcirc'] 
 
 
+
+def promotetoarray(l):
+    if isnumber(l):
+        return array([l])
+    elif isinstance(l, list) or isinstance(l, tuple):
+        return array(l)
+    elif isinstance(l, ndarray):
+        return l
+    raise OptimaException("Expecting a number/list/tuple/ndarray sequence: %s" % repr(l))
```

Comment:
no problems with that. in future, I'd probably centralise error-checking like this throughout the codebase.